checkov/terraform/graph_builder/local_graph.py

```diff
@@ -500,6 +502,52 @@ def get_abspath(self, path: str) -> str:
             self.abspath_cache[path] = dir_name
         return dir_name
 
+    @staticmethod
+    def get_current_block_address(vertex: TerraformBlock) -> str:
+        current_address = ''
+        if vertex.block_type == BlockType.RESOURCE:
+            current_address = vertex.name
+        if vertex.block_type == BlockType.MODULE:
+            current_address = f'{BlockType.MODULE}.{vertex.name}'
+        return current_address
+
+    def update_nested_module_address_recursive(self, edge: Edge) -> str:
+        vertex = self.vertices[edge.origin]
+        if vertex.attributes.get(CustomAttributes.TF_RESOURCE_ADDRESS):
+            return vertex.attributes.get(CustomAttributes.TF_RESOURCE_ADDRESS)
+
+        current_address = TerraformLocalGraph.get_current_block_address(vertex)
+
+        new_edge = None
+        for cur_edge in self.edges:
+            if cur_edge.origin == edge.dest:
+                new_edge = cur_edge
+                break
+
+        # This is the first module in the chain
+        if new_edge is None:
+            current_address = f'{vertex.block_type}.{vertex.name}'
+            self.vertices[edge.origin].attributes[CustomAttributes.TF_RESOURCE_ADDRESS] = current_address
+            return current_address
+
+        nested_address = self.update_nested_module_address_recursive(new_edge)
```

Comment:
what will happen if the edge is to a resource? (via variable or local) we will get an address of `resource_type1.resource1.resource_type2.resource2`